assets/js/src/newsletters/send/date_time.jsx

```diff
@@ -13,8 +13,15 @@ class DateTime extends React.Component {
     this.state = this.buildStateFromProps(props);
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.setState(this.buildStateFromProps(nextProps));
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.value !== prevProps.value
+      || this.props.defaultDateTime !== prevProps.defaultDateTime
+    ) {
+      setImmediate(() => {
```

Comment:
Is this `setImmediate` here just to stop ESLint from complaining about using `setState`? If so then why to use a comment to make it ignore this statement?

Reply:
That is not the only reason. The other reason is to leave react a little bit of space to finish rendering before doing the second render after the state is changed. So really just to make it more asynchronous.